### PR TITLE
Add housefilter

### PR DESCRIPTION
**Pull request by luziehtan**

### CODE REVIEW

File: src/components/HouseFilter/HouseFilter.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  filterGryffindor.textContent = 'Gryffindor'
+  filterGryffindor.addEventListener('click', () => {
+    const charactersGryffindor = characters.filter(
+      character => character.house === 'Gryffindor'
+    )
+    console.log(charactersGryffindor)
+    createCards(charactersGryffindor)
+  })
+  document.body.append(filterGryffindor)
+}
```
Comment (dw0111) [inline src/components/HouseFilter/HouseFilter.js]:
```suggestion
export default function HouseFilter(createCards) {

const characters = []

getCharacters()
    .then(people => characters.push(...people))
    .catch(error => handleGetCharacterError(error))

  const filterGryffindor = createElement('button', {textContent: 'Gryffindor'})
  filterGryffindor.addEventListener('click', () => {
    const charactersGryffindor = characters.filter(
      character => character.house === 'Gryffindor'
    )
    console.log(charactersGryffindor)
    createCards(charactersGryffindor)
  })
  document.body.append(filterGryffindor)
}
```

File: src/components/HouseFilter/HouseFilter.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
+.HouseFilter__box {
```
Comment (macfrei) [inline src/components/HouseFilter/HouseFilter.css:1]: Love your BEM!

File: src/components/HouseFilter/HouseFilter.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    className: 'HouseFilter__box',
+  })
+  document.body.append(filterButtonBox)
+
+  const characters = []
+  getCharacters()
+    .then(people => characters.push(...people))
+    .catch(error => console.log(error))
+
+  const filterGryffindor = FilterButton('Gryffindor', characters, createCards) // house, characters, createCards
```
Comment (macfrei) [inline src/components/HouseFilter/HouseFilter.js:22]: Really good refactor :)

**Review by macfrei — COMMENTED** [review]
Well done! 🚀

**Pull request merged**